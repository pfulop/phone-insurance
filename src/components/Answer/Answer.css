.wrapper {
  margin: 16px 0;
}

.editButton {
  background-color: #4e8098;
  text-decoration: none;
  text-transform: none;
  color: white;
  padding: 4px 16px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.2s;
  border-radius: 4px;
  border: none;
  font-size: 24px;
  margin-right: 0;
  margin-left: auto;
}

.editButton:hover {
  box-shadow: 0 0px 0px 0 rgba(0, 0, 0, 0.2);
}

.answer {
  font-weight: bold;
}

.question {
  text-align: left;
  font-style: italic;
}
