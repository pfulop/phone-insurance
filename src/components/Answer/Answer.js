import React from 'react';
import css from './Answer.css';

export default ({ answer, val, toQuestion }) =>
  val ? (
    <div className={css.wrapper}>
      <div className={css.question}> {answer.text} </div>
      <div className={css.answer}>{val}</div>
      {toQuestion && (
        <button onClick={toQuestion} type="button" className={css.editButton}>
          Edit
        </button>
      )}
    </div>
  ) : (
    <div />
  );
