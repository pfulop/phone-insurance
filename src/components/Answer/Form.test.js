import React from 'react';
import Answer from './Answer';
import { shallow } from 'enzyme';

describe('Answer', () => {
  it('should be defined', () => {
    expect(Answer).toBeDefined();
  });
  it('should render correctly without function', () => {
    const tree = shallow(
      <Answer
        answer={{
          text: 'Are you sure ?',
        }}
        val={'yes'}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly with function', () => {
    const tree = shallow(
      <Answer
        answer={{
          text: 'Are you sure ?',
        }}
        val={'yes'}
        toQuestion={() => {}}
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
