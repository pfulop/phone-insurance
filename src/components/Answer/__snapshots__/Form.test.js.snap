// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer should render correctly with function 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    answer={
        Object {
            "text": "Are you sure ?",
          }
    }
    toQuestion={[Function]}
    val="yes"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
           
          Are you sure ?
           
</div>,
        <div
          className={undefined}
>
          yes
</div>,
        <button
          className={undefined}
          onClick={[Function]}
          type="button"
>
          Edit
</button>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Are you sure ?",
            " ",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Are you sure ?",
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "yes",
          "className": undefined,
        },
        "ref": null,
        "rendered": "yes",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Edit",
          "className": undefined,
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Edit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
             
            Are you sure ?
             
</div>,
          <div
            className={undefined}
>
            yes
</div>,
          <button
            className={undefined}
            onClick={[Function]}
            type="button"
>
            Edit
</button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Are you sure ?",
              " ",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Are you sure ?",
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "yes",
            "className": undefined,
          },
          "ref": null,
          "rendered": "yes",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit",
            "className": undefined,
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Edit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Answer should render correctly without function 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    answer={
        Object {
            "text": "Are you sure ?",
          }
    }
    val="yes"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}
>
           
          Are you sure ?
           
</div>,
        <div
          className={undefined}
>
          yes
</div>,
        undefined,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Are you sure ?",
            " ",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Are you sure ?",
          " ",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "yes",
          "className": undefined,
        },
        "ref": null,
        "rendered": "yes",
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}
>
             
            Are you sure ?
             
</div>,
          <div
            className={undefined}
>
            yes
</div>,
          undefined,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Are you sure ?",
              " ",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Are you sure ?",
            " ",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "yes",
            "className": undefined,
          },
          "ref": null,
          "rendered": "yes",
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
